apache/ambari
# AMBARI-25130 - Infra Manager: backport fixes from 2.8.0

## What changes were proposed in this pull request?

apply fixes to branch-2.7:
https://github.com/apache/ambari-infra/pull/23
https://github.com/apache/ambari-infra/pull/21
https://github.com/apache/ambari-infra/pull/20
https://github.com/apache/ambari-infra/pull/19
https://github.com/apache/ambari-infra/pull/18
https://github.com/apache/ambari-infra/pull/17
https://github.com/apache/ambari-infra/pull/16
https://github.com/apache/ambari-infra/pull/15
https://github.com/apache/ambari-infra/pull/14
https://github.com/apache/ambari-infra/pull/13
https://github.com/apache/ambari-infra/pull/12
https://github.com/apache/ambari-infra/pull/11
https://github.com/apache/ambari-infra/pull/10
https://github.com/apache/ambari-infra/pull/9
https://github.com/apache/ambari-infra/pull/8
https://github.com/apache/ambari-infra/pull/7
https://github.com/apache/ambari-infra/pull/6
https://github.com/apache/ambari-infra/pull/4
## How was this patch tested?

ITs passed

manually:
1. deploy ambari and a cluster: zk, hdfs, infra-solr, logsearch
2. install infra manager to one of the host
3. set the zk connection string of `archive-service-logs` job in infra manager properties file
4. copy `hdfs-site.xml` and `core-site.xml` to infra managers conf directory
5. create `/archives/service-logs` folder on hdfs and set as a destination folder in infra manager properties file of the `archive-service-logs` job
6. allow write into `/archives/service-logs` folder for all users
7. start infra manager
8. start `archive-service-logs` job
9. check `/archives/service-logs` folder on hdfs for archived logs
